pkg/commands/git_commands/repo_paths.go

```diff
@@ -64,9 +63,9 @@ func (self *RepoPaths) IsBareRepo() bool {
 func MockRepoPaths(currentPath string) *RepoPaths {
 	return &RepoPaths{
 		worktreePath:       currentPath,
-		worktreeGitDirPath: path.Join(currentPath, ".git"),
+		worktreeGitDirPath: filepath.Join(currentPath, ".git"),
```

Comment:
I disagree; we should be using "correct" paths everywhere, and the filepath package is the easy way to achieve that without having to use platform-conditional code everywhere. I noticed this because I tested editing a repo-specific config file on Windows, and upon hitting `e` it showed me a dialog containing `C:\some\repo/.git/lazygit.yml`, which works, but looks very bad.

The only place where we need conditional code is the `repo_paths_test.go` test; that test is now quite awkward I admit, but I still think it's better, because previously the paths used there didn't make any sense on Windows.

Reply:
I feel like the blast radius is larger than that: pkg/commands/git_commands/worktree_loader.go hard-codes '/' as a file path separator, as does pkg/gui/presentation/files.go. If we want to be using platform-specific path separators everywhere, we should update those files too, would you agree?

Also, we should keep in mind that we want integration tests to work (some day) on windows, which means being able to force lazygit to use a specific path separator for the sake of keeping conditional logic out of our test assertions.

Reply:
> I feel like the blast radius is larger than that: pkg/commands/git_commands/worktree_loader.go hard-codes '/' as a file path separator, as does pkg/gui/presentation/files.go. If we want to be using platform-specific path separators everywhere, we should update those files too, would you agree?

Wow, that makes it trickier. The places that you are pointing out are dealing with relative paths (relative to the repo root), and I'd say those should keep using `/`; even git itself outputs them using `/`, e.g. in `git status -s`. I suppose we should use backslashes only when dealing with absolute paths.

From what I can see, all the places I touched in this commit are dealing with absolute paths, so I'd propose to keep this, and not change the other ones. Does that seem reasonable?

> Also, we should keep in mind that we want integration tests to work (some day) on windows, which means being able to force lazygit to use a specific path separator for the sake of keeping conditional logic out of our test assertions.

I don't see any places in our integration test assertions that expect a path, do you have an example?

I'm more worried about unit tests; it would be bad if we need more ugliness of the kind that I had to add in `repo_paths_test.go` in this commit.

Reply:
> From what I can see, all the places I touched in this commit are dealing with absolute paths, so I'd propose to keep this, and not change the other ones. Does that seem reasonable?

Yeah that's reasonable

> I don't see any places in our integration test assertions that expect a path, do you have an example?

I was thinking about any assertions involving the files panel (when not in tree mode). But as you say, those are relative paths so we don't need to worry about them.

> I'm more worried about unit tests; it would be bad if we need more ugliness of the kind that I had to add in repo_paths_test.go in this commit.

Agreed. Though I think if we only encounter such awkwardness with absolute paths, it may not be so bad.